### PR TITLE
ACORN-343: logo multi language

### PR DESCRIPTION
**Pull request by abolabo**

### CODE REVIEW

File: public_html/admin/controller/common/head.php [hunk 1]
```diff
@@ -60,8 +62,15 @@ public function main()
...
         }
-
-        $this->data['icon'] = $this->config->get('config_icon');
+        if(is_numeric($this->config->get('config_icon'))){
+            $r = new AResource('image');
+            $resourceInfo = $r->getResource($this->config->get('config_icon'), $this->language->getLanguageID());
+            if($resourceInfo){
+                $this->data['icon'] = $resourceInfo['type_dir'].$resourceInfo['resource_path'];
+            }
+        }else {
```
Comment (abantecart) [inline public_html/admin/controller/common/head.php:71]: Need space ->  } else {

File: public_html/admin/controller/responses/listing_grid/customer.php [hunk 2]
```diff
@@ -170,13 +170,14 @@ public function update()
...
             $error = new AError('');
-            return $error->toJSONResponse(
+            $error->toJSONResponse(
                 'NO_PERMISSIONS_402',
                 [
                     'error_text'  => sprintf($this->language->get('error_permission_modify'), 'listing_grid/customer'),
                     'reset_value' => true,
                 ]
             );
+            return;
```
Comment (abantecart) [inline public_html/admin/controller/responses/listing_grid/customer.php:180]: Why error is no longer returned? Validation should return the error. At least validation has to return FALSE if failed.

File: public_html/admin/controller/responses/listing_grid/customer.php [hunk 3]
```diff
@@ -201,13 +202,14 @@ public function update()
...
                             $error = new AError('');
-                            return $error->toJSONResponse(
+                            $error->toJSONResponse(
                                 'VALIDATION_ERROR_406',
                                 [
                                     'error_text'  => $err,
                                     'reset_value' => false,
                                 ]
                             );
+                            return;
```
Comment (abantecart) [inline public_html/admin/controller/responses/listing_grid/customer.php:212]: Same as above

File: public_html/admin/controller/responses/listing_grid/customer.php [hunk 4]
```diff
@@ -221,13 +223,14 @@ public function update()
...
```
Comment (abantecart) [inline public_html/admin/controller/responses/listing_grid/customer.php:233]: same

File: public_html/admin/controller/responses/setting/setting_quick_form.php [hunk 5]
```diff
@@ -240,40 +241,30 @@ public function quick_start_save_next()
...
-                        //we save resource ID vs resource path
-                        $this->request->post['config_logo'] = $this->request->post['config_logo_resource_id'];
+                                ? 'appearance'
+                                : $this->request->get['tmpl_id'];
+
+                    foreach (['config_logo', 'config_mail_logo', 'config_icon'] as $n) {
+                        $newName = in_array($n, ['config_logo', 'config_mail_logo']) ? $n.'_'.$languageId : $n;
+                        if (isset($post[$n.'_resource_id'])) {
+                            //we save resource ID vs resource path
+                            $post[$newName] = $post[$n.'_resource_id'];
```
Comment (abantecart) [inline public_html/admin/controller/responses/setting/setting_quick_form.php:256]: 1. We should not change POST. We talked about it. Even if we have it in the old code, we should go away from this. 
2. I do not think global post will be changed as this is a local copy.

File: public_html/admin/controller/responses/setting/setting_quick_form.php [hunk 6]
```diff
@@ -240,40 +241,30 @@ public function quick_start_save_next()
...
-                    ) {
-                        //we save resource ID vs resource path
-                        $this->request->post['config_icon'] = $this->request->post['config_icon_resource_id'];
-                    }
-                    $this->model_setting_setting->editSetting($section, $this->request->post, $store_id);
-                } else {
-                    $this->model_setting_setting->editSetting($group, $this->request->post, $store_id);
                 }
+
+                $this->model_setting_setting->editSetting($section, $post, $store_id);
```
Comment (abantecart) [inline public_html/admin/controller/responses/setting/setting_quick_form.php:266]: Was this update confirmed by user? Need a code comment here. Why do we update and what we update?

File: public_html/admin/model/sale/customer.php [hunk 7]
```diff
@@ -1142,41 +1172,25 @@ public function sendApproveMail($customer_id)
...
                 $store_info['store_url'] = $store_info['config_url'].'index.php?rt=account/login';
             } else {
                 $store_info = [
-                    'store_name'       => $this->config->get('store_name'),
-                    'store_url'        => $this->config->get('config_url').'index.php?rt=account/login',
-                    'config_mail_logo' => $this->config->get('config_mail_logo'),
-                    'config_logo'      => $this->config->get('config_logo'),
+                    'store_name'                    => $this->config->get('store_name'),
+                    'store_url'                     => $this->config->get('config_url').'index.php?rt=account/login',
+                    'config_mail_logo_'.$languageId => $this->config->get('config_mail_logo_'.$languageId),
```
Comment (abantecart) [inline public_html/admin/model/sale/customer.php:1183]: It is not a good idea to have dynamic key in the array. Not clear what to look for.

File: public_html/core/helper/utils.php [hunk 8]
```diff
@@ -1354,4 +1354,45 @@ function setCookieOrParams($name = null, $value = null, $options = [])
...
     }
+}
+
+/**
+ * @param string|int $source
+ *
+ * @return array
+ * @throws AException
+ */
+function getMailLogoDetails( &$source)
```
Comment (abantecart) [inline public_html/core/helper/utils.php:1365]: getMailLogo is a sufficient name.

File: public_html/install/abantecart_database.sql [hunk 9]
```diff
@@ -1670,8 +1670,8 @@ INSERT INTO `ac_settings` (`group`, `key`, `value`) VALUES
 -- Appearance
 
 ('appearance','storefront_width','100%'),
-('appearance','config_logo','image/18/73/3.png'),
-('appearance','config_mail_logo','image/18/73/3.png'),
+('appearance','config_logo_1','image/18/73/3.png'),
```
Comment (abantecart) [inline public_html/install/abantecart_database.sql:1673]: I do not like that. Default should stay as it was before. 
1. People will loose logos on upgrade. 
2. It is not clear and guarantied that default will be 1

**Review by abantecart — CHANGES_REQUESTED** [review]

File: public_html/admin/controller/responses/listing_grid/customer.php (hunk 2 above)
Comment (abolabo) [reply public_html/admin/controller/responses/listing_grid/customer.php:180]: it's no a validation. Function name is $error->toJSONResponse

File: public_html/admin/controller/responses/listing_grid/customer.php (hunk 3 above)
Comment (abolabo) [reply public_html/admin/controller/responses/listing_grid/customer.php:212]: see answer above

File: public_html/admin/controller/responses/setting/setting_quick_form.php (hunk 5 above)
Comment (abolabo) [reply public_html/admin/controller/responses/setting/setting_quick_form.php:256]: 1. post is not changing here. see line 244. $post variable is not a reference. it's a separate var!
2. see above!

File: public_html/admin/controller/responses/listing_grid/customer.php (hunk 2 above)
Comment (abolabo) [reply public_html/admin/controller/responses/listing_grid/customer.php:180]: it's a void function. It send response as json data directly to php output. No return data needed here.

File: public_html/install/abantecart_database.sql (hunk 9 above)
Comment (abolabo) [reply public_html/install/abantecart_database.sql:1673]: upgrade script will write later. It's a installation sql.

**Comment by abantecart** [comment]
Why was this merged?

File: public_html/install/abantecart_database.sql (hunk 9 above)
Comment (abantecart) [reply public_html/install/abantecart_database.sql:1673]: It has to stay as it was with default values.

**Pull request merged**